starboard/android/shared/media_codec_bridge.cc

```diff
@@ -348,6 +359,15 @@ MediaCodecBridge::~MediaCodecBridge() {
     return;
   }
 
+  if (MediaCodecBridgeEradicator::GetInstance()->IsEnabled()) {
+    if (MediaCodecBridgeEradicator::GetInstance()->Destroy(
+            j_media_codec_bridge_, j_reused_get_output_format_result_)) {
+      return;
+    }
+    SB_LOG(WARNING)
+        << "MediaCodecBridge destructor fallback into none radicator mode.";
+  }
+
   JniEnvExt* env = JniEnvExt::Get();
 
   env->CallVoidMethodOrAbort(j_media_codec_bridge_, "stop", "()V");
```

Comment:
We should also consider resetting `mNativeMediaCodecBridge` (https://github.com/youtube/cobalt/blob/24.lts.1%2B/starboard/android/apk/app/src/main/java/dev/cobalt/media/MediaCodecBridge.java#L899C7-L899C30), and revisit the Java MediaCodecBridge implementation to ensure that it won't use any resources destroyed earlier than it.

Reply:
It has synchronized (this) to protect mNativeMediaCodecBridge, it is already thread-safe. Did I miss anything?